packages/mask/.webpack/config.ts

```diff
@@ -175,7 +175,7 @@ export function createConfiguration(_inputFlags: BuildFlags): Configuration {
             })(),
             new EnvironmentPlugin({
                 NODE_ENV: flags.mode,
-                shadowRootMode: flags.devtools ? 'open' : 'closed',
+                shadowRootMode: flags.devtools || flags.channel === 'e2e' ? 'open' : 'closed',
```

Comment:
what if we let beta&nightly versions have opened shadow root?

Reply:
then we can avoid two more channels

Reply:
Done here: https://github.com/DimensionDev/Maskbook/pull/8130